controllers/database.go

```diff
@@ -38,6 +39,10 @@ var dbTemplates = []string{
 
 // extract to var for mocking in testing
 var ConnectAndQueryDatabase = func(host, port, username, password, dbname string) bool {
+	// Create a context with a timeout of 1 second
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
is a 1 second timeout too aggressive?  Perhaps this should be configurable?

Reply:
updated